src/tss2-esys/api/Esys_StartAuthSession.c

```diff
@@ -450,10 +450,6 @@ Esys_StartAuthSession_Finish(
             secret_size += keyHash_size;
         if (bind != ESYS_TR_NONE && bindNode != NULL)
             secret_size += bindNode->auth.size;
-        if (secret_size == 0) {
-            return_error(TSS2_ESYS_RC_GENERAL_FAILURE,
-                         "Invalid secret size (0).");
-        }
```

Comment:
The C spec [1] in 7.20.3 says:
"If the size of the space requested is zero, the behavior is implementation defined: either a null pointer is returned, or the behavior is as if the size were some nonzero value, except that the returned pointer shall not be used to access an object."
So if the size is zero the returned 'secret' needs special handling.
[1] http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf